### ISSUE TITLE
RESIDENT EVIL 2 / BIOHAZARD RE:2 "1-Shot Demo" does not start (961440)

### ISSUE THREAD

**Issue by laichiaheng** [ValveSoftware/Proton#2210]
# Compatibility Report
- Name of the game with compatibility issues: RESIDENT EVIL 2 / BIOHAZARD RE:2 "1-Shot Demo" 
- Steam AppID of the game: 961440

## System Information
- GPU: RX 460
- Driver/LLVM version: Mesa 19.0/7.0.0
- Kernel version: 4.20
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: 3.7-8

## I confirm:
- [V] that I haven't found an existing compatibility report for this game.
- [V] that I have checked whether there are updates for my system available.

[steam-961440.log](https://github.com/ValveSoftware/Proton/files/2751771/steam-961440.log)

## Symptoms 

I can't start the game

**Comment by guihkx** [comment]
## System Information
- GPU: GTX 660
- Driver/LLVM version: NVIDIA 415.25
- Kernel version: 4.20.0
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.githubusercontent.com/guihkx/e6b6d755669555ad51131e7545504c71/raw/92fe25517f8a514ed451ecc4497ece7a2720c529/gistfile1.txt
- Proton version: 3.16-6 Beta



## Symptoms

Game launches into a black screen for a few seconds and then closes.

## Reproduction

Running the game with no launch options (except `PROTON_LOG=1`), gives me this blank window, which then closes itself a few seconds later [(LOG FILE HERE)](https://github.com/ValveSoftware/Proton/files/2751811/steam-961440.log):

![black-screen-res2-demo](https://user-images.githubusercontent.com/626206/51070815-92e07100-162e-11e9-972c-abb265a081bf.png)


The same happens when I try to launch the game with `PROTON_USE_WINED3D=1`, the only difference being this error that pops up [(LOG FILE HERE)](https://github.com/ValveSoftware/Proton/files/2751813/steam-961440.log):

![image](https://user-images.githubusercontent.com/626206/51070858-46e1fc00-162f-11e9-8253-660f66120237.png)

**Comment by laichiaheng** [comment]
@kisak-valve Does the label mean it won't be fixed?

**Comment by doitsujin** [comment]
This is fixed in DXVK as of https://github.com/doitsujin/dxvk/commit/bee21e753977ffc046303f5b98fb8e54c7f45727.

Please note that the game does not work on RADV and AMDVLK at the moment.

**Comment by laichiaheng** [comment]
@doitsujin Where should I note it?
So it is playable on RADV and AMDVLK after this fix?

**Comment by doitsujin** [comment]
No, that's why I mentioned that it *doesn't* work on those drivers. It will run, but not render correctly.

FWIW I think the reason why it renders incorrectly on RADV is a broken shader in the game. This is not really somthing that can be fixed in Proton / DXVK.

**Comment by guihkx** [comment]
Okay so it runs with latest DXVK (thanks!), the performance is what I already had expected on my ancient GTX 660, but there's this weird bug that keeps turning the game darker over time, even with brightness settings maxed out:

https://www.youtube.com/watch?v=0dlqfwPHYyg&feature=youtu.be

BTW I know the frame times and the fps are shit, but tbh I didn't even try tweaking the game graphics much. Might try doing that later...

EDIT: Also, apparently the game can't figure out how much VRAM I have:

![20190112192304_1](https://user-images.githubusercontent.com/626206/51078917-90206300-16a4-11e9-84dd-8dbc186e79ca.jpg)
![20190112191846_1](https://user-images.githubusercontent.com/626206/51078918-9282bd00-16a4-11e9-8518-1b29691be70f.jpg)

**Comment by doitsujin** [comment]
@guihkx 
> but there's this weird bug keeps turning the game darker over time, even with brightness settings maxed out:

That appears to be another game bug, see https://steamcommunity.com/app/961440/discussions/0/2521353993639265147/

**Comment by laichiaheng** [comment]
@doitsujin Where should I report the bug? Whom should I report? The mesa developers?

**Comment by doitsujin** [comment]
It's not a Mesa bug, it's a game bug. The best thing would be to get Capcom to fix it, but i highly doubt that's going to happen.

**Comment by jollycar** [comment]
The game starts and runs with Wine 4 rc5 and DXVK 0.95, but the game gets bugged out when you try to access the laptop/computer in the game. You don't get an interface where you access the computer, instead you can just walk around without the possibility to interact with the environment, not even escape works anymore and I have to forcefully quit the game (alt-f4). Anyway: the performance is not bad and great to see progression

**Comment by laichiaheng** [comment]
@ivohuisjes How do you open a Steam game via Wine? Install a Windows version Steam client?
I have wine-staging and DXVK 0.95 installed.

**Comment by jollycar** [comment]
Lutris > Manage runners > Wine Steam > Configure > Wine version: 4 RC5 (staging) > (check) Enable DXVK 0.95

Note: The latest Lutris version (I use 0.4.23) will automatically detect the latest available DXVK version.
Note: To install RE2 demo in Lutris, just add a new wine steam game and put in the app id: 961440

This may help: https://www.linuxuprising.com/2018/09/how-to-use-lutris-to-play-windows-games.html

**Comment by laichiaheng** [comment]
@ivohuisjes But there is no 0.95 in the DXVK select menu.

**Comment by laichiaheng** [comment]
@ivohuisjes Did I do it right?
```
export WINEPREFIX=~/.local/share/lutris/runners/winesteam/prefix64/
winetricks --force setup_dxvk.verb
```
There is no way to download RE2 1 shot demo on it, because I can't browse the website in Wine Steam

**Comment by guihkx** [comment]
>That appears to be another game bug, see https://steamcommunity.com/app/961440/discussions/0/2521353993639265147/

Yep, it definitely is. I've booted onto my Windows 10 installation (it's on the same machine) and the exact same thing happens. Too bad, I highly doubt they're fixing the demo.

**Comment by ghost** [comment]
@doitsujin 
> No, that's why I mentioned that it _doesn't_ work on those drivers. It will run, but not render correctly.
> 
> FWIW I think the reason why it renders incorrectly on RADV is a broken shader in the game. This is not really somthing that can be fixed in Proton / DXVK.

Can you explain? How bad is this incorrect rendering, and if it's a game bug, why can it work on Nvidia and AMD on Windows but not mesa, if it's not a mesa problem?

**Comment by doitsujin** [comment]
@z0z0z RADV optimizes a load instruction with an undefined result away, the proprietary driver doesn't. Even so, the reason why it works on those drivers basically comes down to sheer luck.

**Comment by laichiaheng** [comment]
Can't Valve update their DXVK version in Proton?

**Comment by doitsujin** [comment]
The invisible object issue on RADV is fixed in Mesa as of [this commit](https://gitlab.freedesktop.org/mesa/mesa/commit/8424cd8fbd1671c4c13f57cfa34bf8145d0fffcf).

**Comment by kisak-valve** [comment]
RE2 1-shot demo Non-Stop Shooting / Judder / Drop Frames Suddenly / Strange Texture (961440)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/2236.
@laichiaheng posted on 2019-01-19T08:08:27:

# Compatibility Report
- Name of the game with compatibility issues: RESIDENT EVIL 2 / BIOHAZARD RE:2 "1-Shot Demo"
- Steam AppID of the game: 961440

## System Information
- GPU: <RX 580>
- Driver/LLVM version: <Mesa 19.0.0-devel (git-6ca652f 2019-01-19 bionic-oibaf-ppa)/LLVM 7.0.1>
- Kernel version: <4.20.3>
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: 3.16-6-BETA with the DXVK0.95 dlls

## I confirm:
- [V] that I haven't found an existing compatibility report for this game.
- [V] that I have checked whether there are updates for my system available.

[steam-961440.log](https://github.com/ValveSoftware/Proton/files/2775338/steam-961440.log)


## Symptoms 
1. non-stop shooting
2. judder
3. drop frames suddenly
4. strange texture

## Reproduction
1. non-stop shooting: I don't know, it happens a lot if I have shot many bullets, when I aim the target, it keeps shooting.
2. judder: walk around.
3. drop frames suddenly: Sometimes I shoot or walk around or into an animation.
4. strange texture:
There is a setting which is called Screen Space Reflection
![2019-01-19 15-49-37](https://user-images.githubusercontent.com/9532105/51424108-6dda9800-1c04-11e9-9c69-549274d2cfdf.png)
If I enable Screen Space Reflection, the pillar looks like this:
![2019-01-19 15-49-02](https://user-images.githubusercontent.com/9532105/51424102-51d6f680-1c04-11e9-8a69-c8cdd3fe920c.png)
If I disable it, it looks normal:
![2019-01-19 15-49-44](https://user-images.githubusercontent.com/9532105/51424105-5dc2b880-1c04-11e9-9599-87aea37df7ed.png)

**Comment by doitsujin** [comment]
@laichiaheng The screen space reflections are not a bug, this is what it's supposed to look like.

**Comment by laichiaheng** [comment]
What! How does it look so crappy and ugly.
By the way, is it normal to see DXVK on Linux performs much worse than DX11 on Windows?
RX580 seems to be able to run at 1080p60fps on Windows 10, but it almost runs at 1080p4Xfps on Ubuntu 18.04.1

**Comment by doitsujin** [comment]
> By the way, is it normal to see DXVK on Linux performs much worse than DX11 on Windows?

Lower performance should be expected, yes. That being said, on my RX 480 the performance difference is at around 15% or so in this game.

**Comment by laichiaheng** [comment]
@doitsujin It was 60fps on AMDGPU-PRO, but 4Xfps on RADV, I didn't expect that there was such huge different in performance with these 2 drivers.
